framework/Models/PostProcessors/BasicStatistics.py

```diff
@@ -998,21 +1017,17 @@ def __runLocal(self, inputData):
         en = targWeight.sum()**2/np.sum(targWeight**2)
         targDa = dataSet[target]
         if self.pivotParameter in targDa.sizes.keys():
-          percentileSte = []
-          for label, group in targDa.groupby(self.pivotParameter):
-            subPercentileSte = []
+          percentileSte = np.zeros((len(self.pivotValue), len(percent))) # array
+          for i, (label, group) in enumerate(targDa.groupby(self.pivotParameter)): # array
             if group.values.min() == group.values.max():
-              # all values are the same
-              for pct in percent:
-                subPercentileSte.append(0.0)
+              subPercentileSte = np.array([0.0]*len(percent))
             else:
               # get KDE
               kde = stats.gaussian_kde(group.values, weights=targWeight)
```

Comment:
We cannot avoid this when there are probability weights present. This is because xarray does not currently support weighted quantiles. Without going too far down the rabbit hole, here are some of the issues being worked to enable that functionality:

[https://github.com/pydata/xarray/issues/1371](https://github.com/pydata/xarray/issues/1371)
[https://github.com/numpy/numpy/issues/8935](https://github.com/numpy/numpy/issues/8935)
[https://github.com/numpy/numpy/pull/9211](https://github.com/numpy/numpy/pull/9211)

This is why we have to handle weighted percentiles on our own, which requires looping over targets and time for percentile and percentile standard error. I have made it so that we no longer have to loop over a third parameter (percent) as well.

Reply:
By switching to the standard normal formulation of the standard error, we no longer have to loop through targets, time, or percent. This speeds up the calculation significantly. I left the KDE implementation commented out in the code in case we want to revisit it.